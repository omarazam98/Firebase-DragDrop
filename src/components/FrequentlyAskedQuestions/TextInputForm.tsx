import React from "react";
import { blue, green } from '@material-ui/core/colors';
import { Button, TextField } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/styles';
import {
  createStyles,
  Theme,
  makeStyles,
  createMuiTheme,
} from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
      flexWrap: 'wrap',
    },
    button: {
      margin: theme.spacing(1),
    },
    margin: {
      margin: theme.spacing(1),
    },
    email: {
      margin: theme.spacing(1),
      width: 250,
    },
  }),
);

const theme = createMuiTheme({
  palette: {
    primary: green,
    secondary: blue,
  },
});

interface TextFormProps {
  state: {
    isQuestionAsked: boolean,
    questionValue: string,
    questionLabel: string,
    prompt: string,
    afterSubmit: string,
    submitLabel: string,
    userEmail: string,
    userEmailLabel: string,
  };
  handleSubmit: any;
  handleChange: any;
  handleBlur: any;
}

const TextInputForm : React.FC<TextFormProps> = (props) => {
  const classes = useStyles();
  return (
    <ThemeProvider theme={theme}>
    <form onSubmit={props.handleSubmit} name="questionText">
      <h3>
        {props.state.prompt}
      </h3>
      <TextField
        id="mui-theme-provider-outlined-input-question"
        label={props.state.questionLabel}
        variant="outlined"
        className={classes.margin}
        multiline
        onChange={props.handleChange}
        onBlur={props.handleBlur}
        value={props.state.questionValue}
      />
      <TextField
        id="mui-theme-provider-outlined-input-email"
        label={props.state.userEmailLabel}
        variant="outlined"
        className={classes.email}
        onChange={props.handleChange}
        onBlur={props.handleBlur}
        value={props.state.userEmail}
      />
      <Button
        variant="contained"
        className={classes.button}
        color="secondary"
        type="submit"
        id="question-submit"
        onClick={props.handleSubmit}>
        {props.state.submitLabel}
      </Button>
      {
        props.state.isQuestionAsked ?
          <p className={classes.margin}>
            {props.state.afterSubmit}
          </p>
          : null
      }
    </form>
    </ThemeProvider>
  );
};
export default TextInputForm;
