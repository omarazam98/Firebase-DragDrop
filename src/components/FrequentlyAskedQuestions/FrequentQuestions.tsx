import React, { Component } from 'react';
import SimpleAccordion from './SimpleAccordion';
import TextInputForm from './TextInputForm';

interface FrequentQuestionsProps {
  questionList :PairType[];
  handleSubmit: any;
}
interface FrequentQuestionsState {
  question: string;
  userEmail: string;
  isQuestionAsked: boolean;
  touched: {};
}
const INITIAL_STATE:FrequentQuestionsState = {
  question: '',
  userEmail: '',
  isQuestionAsked: false,
  touched: {},
}

interface PairType {
  question: string;
  answer: string;
}

export class FrequentQuestions extends Component<FrequentQuestionsProps, FrequentQuestionsState> {
  constructor (props) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleBlur = this.handleBlur.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.state = INITIAL_STATE;
  }

  handleSubmit (event) {
    event.preventDefault();
    try {
      if (this.state.userEmail !== '') {
        this.props.handleSubmit(this.state.question, this.state.userEmail);
      } else {
        this.props.handleSubmit(this.state.question);
      }
    } catch (error) {
      console.log(error);
    } finally {
      this.setState(prevState => ({
        isQuestionAsked: true,
        question: '',
        userEmail: '',
      }));
    }
  }

  handleBlur (event) {
    const target = event.target;
    const name = target.name;
    this.setState(prevState =>
      ({
        touched: {
          ...prevState.touched,
          [name]: true,
        },
      }));
  }

  handleChange (event) {
    const target = event.target;
    const value = event.target.value;
    target.id === 'mui-theme-provider-outlined-input-question' ?
      this.setState(prevState => ({
        question: value,
      }))
      :
      this.setState(prevState => ({
        userEmail: value,
      }));
  }

  render () {
    const textInputState = {
      userEmail: this.state.userEmail,
      questionValue: this.state.question,
      isQuestionAsked: this.state.isQuestionAsked,
      userEmailLabel: 'Your Email (Not Required)',
      questionLabel: 'Ask us your question',
      prompt: 'Don\'t see your question? feel free to ask ' +
        'us whatever you would like and we will add it to our list',
      afterSubmit: 'Thank you for asking your question, we will be ' +
        'sure to get back to you very soon',
      submitLabel: 'Ask Question',
    }
    return(
      <div>
        <h1>Frequently Asked Questions</h1>
        {this.props.questionList.map((pair, index) => (
          <SimpleAccordion
            title={pair.question}
            body={pair.answer}
            key={index}
            style={{ maxWidth:300 }} />
        ))}
        <TextInputForm
          state={{ ...textInputState }}
          handleSubmit={this.handleSubmit}
          handleBlur={this.handleBlur}
          handleChange={this.handleChange}
        />
      </div>
    );
  }
}

export default FrequentQuestions;
