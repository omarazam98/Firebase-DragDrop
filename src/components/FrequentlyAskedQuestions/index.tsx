import React, { useEffect, useState } from 'react';
import { withAPI } from '@winwin/api-firebase';
import FrequentQuestions from './FrequentQuestions';

interface FrequentlyAskedQuestionsPageProps {
  api: any;
}

const FrequentlyAskedQuestionsPage: React.FC<FrequentlyAskedQuestionsPageProps> = (props) => {
  const [questionList, setQuestionList] = useState([]);

  useEffect(() => {
    const frequentQuestionsCollection = props.api.data.faq.getAll();
    frequentQuestionsCollection.then(questions =>
      setQuestionList(questionList => questions));
  },        [props.api.data.faq]);

  const handleSubmit = function (question:string, email?:string) {
    props.api.data.faq.addQuestion(question, email);
  };

  return (
    <div>
      <FrequentQuestions questionList={questionList} handleSubmit={handleSubmit}/>
    </div>
  );

};

export default withAPI(FrequentlyAskedQuestionsPage);
