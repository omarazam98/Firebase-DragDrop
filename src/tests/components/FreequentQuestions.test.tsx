import { shallow, mount } from 'enzyme';
import  { FrequentQuestions } from '../../components/FrequentlyAskedQuestions/FrequentQuestions';
import { frequentQuestionsList } from '../fixtures/frequentQuestions';
import * as React from 'react';

let handleSubmit;
beforeEach(() => {
  handleSubmit = jest.fn();
});

test('Should render frequently asked questions with basic question list', async () => {
  const wrapper = shallow(<FrequentQuestions
    questionList={frequentQuestionsList}
    handleSubmit={handleSubmit} />);
  expect(wrapper).toMatchSnapshot();
  const numQuestions = wrapper.find('SimpleAccordion').length;
  expect(numQuestions).toBe(16);
});

test('Should submit question with inserted question text', async () => {
  const wrapper = mount(<FrequentQuestions
    questionList={frequentQuestionsList}
    handleSubmit={handleSubmit} />);
  const textField = wrapper.find('#mui-theme-provider-outlined-input-question').first();
  await(textField.props().onChange(
    { target: { value: 'Test question?', id:'mui-theme-provider-outlined-input-question' } },
    ));
  console.log(wrapper);
  const e = { preventDefault: jest.fn() };
  wrapper.find('form').first().simulate('submit', e);
  expect(handleSubmit).toHaveBeenCalledWith('Test question?');
});
