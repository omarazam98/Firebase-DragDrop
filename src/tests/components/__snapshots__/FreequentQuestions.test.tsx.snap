// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render frequently asked questions with basic question list 1`] = `
<div>
  <h1>
    Frequently Asked Questions
  </h1>
  <SimpleAccordion
    body="Renting with Win-Win doesn’t mean you’ll be a caretaker responsible for the homeowner’s health and tasks related to their well-being. It simply means you’ll take on chores that will help them stay in their home for longer. You will select which chores you would like to do as part of the intake process.
TLDR: Not care-giving, but helping with care."
    key="0"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What are the responsibilities?"
  />
  <SimpleAccordion
    body="Finding a roommate to get along with can be difficult, but our matching service differentiates us from other rental options. We are dedicated to working on a match that fits both parties, but sometimes not all matches are made in heaven. Each issue will be dealt on a case-by-case basis. Win Win will help facilitate whatever is required whether it be to help housemates work through issues or to shorten the duration of the home share."
    key="1"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What happens if I don’t get along with my seniors?"
  />
  <SimpleAccordion
    body="Studying is demanding and can affect your ability to fulfill assigned duties. If your exam schedule is too demanding to upkeep with your chores, update the app and adjust the tasks list for the time you’ll be unavailable. Not upkeeping your chores with communication will result in fees, so make sure you’re clear on your ability to upkeep your chores."
    key="2"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What about exams?"
  />
  <SimpleAccordion
    body="Home-sharing is not care-giving. Students are only expected to keep up with the agreed-upon chores and companionship, not for the ongoing care of the seniors they are living with.
TLDR: No."
    key="3"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="Am I liable for the health and well-being of the senior?"
  />
  <SimpleAccordion
    body="Although you aren’t being paid an hourly wage, you are being rewarded for your help with significantly discounted rent. That means fewer hours working a part-time job to focus on studies, friends, and enjoying your time at school.
TLDR: No, but your rent is subsidized."
    key="4"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="Am I being paid?"
  />
  <SimpleAccordion
    body="No, you do not need a vehicle."
    key="5"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="Do I need a vehicle?"
  />
  <SimpleAccordion
    body="This depends on how many you have agreed upon completing. The average is 8-10 hours of help around the house a month.
TLDR: You choose how much work you’d like to commit to."
    key="6"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="How many chores/hours a week?"
  />
  <SimpleAccordion
    body="Your safety matters. We ensure background checks on all homeowners and renters. We also match you with ideal candidates best suited for both your home and lifestyle. Our matching process is based on common values, desired chores, interests, hobbies, and habits."
    key="7"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="How does the matching work?"
  />
  <SimpleAccordion
    body="Our prices will always be a fraction of the costs of renting a room in the same area. Our prices are based on the location, the furnishings provided, the quality of the home, and other factors. On top of that, the price is tied into the number of chores you are willing to take on, so if you’re looking for really cheap rent, consider staying open to providing more help around the house.
TLDR: it’s cheap– but how cheap is up to you."
    key="8"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="How much does it cost me for rent?"
  />
  <SimpleAccordion
    body="Paying rent is easy with Win-Win. Through our system, we ensure homeowners are getting paid by taking payments from the renters. This allows us to track what has been paid/when, and that there is no delay in payment.
Simply sign in and to check your balances, or set up recurring payments."
    key="9"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="How do I pay rent?"
  />
  <SimpleAccordion
    body="Busy schedules are a trademark of university life, but making time to attend to chores is necessary and part of a written agreement. Having an open conversation about your schedule and time at home is common courtesy for home-sharing."
    key="10"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What if I’m not home every night?"
  />
  <SimpleAccordion
    body="Although you do not receive discounts for being any kind of student, placing your education on your profile will be a great way to introduce yourself to potential homesharers. Also in certain fields doing this program will be a great way to gain and show your experience working in the field of care."
    key="11"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="Do I get a discount if I’m a medical student?"
  />
  <SimpleAccordion
    body="You are responsible for the chores you have agreed to do on a daily, weekly, or monthly basis. The more chores you agree to help with, the cheaper your rent will be."
    key="12"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What will I be responsible for?"
  />
  <SimpleAccordion
    body="As long as you are a student, you may rent with Win-Win Homesharing. The shortest rentals available are two-months long, although popular leases span between 4, 8, or 12 month periods. All contracts are extendable for as long as both parties are satisfied and want to continue."
    key="13"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="How long can I rent for?"
  />
  <SimpleAccordion
    body="Rooms are furnished and displayed on our platform. Each home is different, so be sure to connect with the senior beforehand to find out more. Sign up and explore today!"
    key="14"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What’s the living space like?"
  />
  <SimpleAccordion
    body="Pets are allowed, but both parties must be notified and willing to live beforehand. Who wouldn’t want to walk a dog for discounted rent? Be sure to list allergies on your profile, if any."
    key="15"
    style={
      Object {
        "maxWidth": 300,
      }
    }
    title="What about pets?"
  />
  <TextInputForm
    handleBlur={[Function]}
    handleChange={[Function]}
    handleSubmit={[Function]}
    state={
      Object {
        "afterSubmit": "Thank you for asking your question, we will be sure to get back to you very soon",
        "isQuestionAsked": false,
        "prompt": "Don't see your question? feel free to ask us whatever you would like and we will add it to our list",
        "questionLabel": "Ask us your question",
        "questionValue": "",
        "submitLabel": "Ask Question",
        "userEmail": "",
        "userEmailLabel": "Your Email (Not Required)",
      }
    }
  />
</div>
`;
